doc/adr/0001-record-architecture-decisions.md

```diff
@@ -0,0 +1,149 @@
+# ADR 1: Record architecture decisions
+
+Date: 2022-06-08
+
+## Status
+
+Draft
+
+## Context
+
+We are in search for a means to document our architectural and design decisions
+for all of our components.
+In order to do that, there is practice called architectural design records, ADR,
+that we can integrate into our workflow.
+
+This does not replace actual architecture documentation, but provides contributors:
+
+* the means to understand architectural and design decisions that were made
+* a framework for proposing changes to the current architecture
+
+For each decision, it is important to consider and remember:
+
+* what we've decided to do,
+* why we made this decision,
+* what is the impact of this decision,
+* what we learnt in the process.
+
+As we're already using rST, Sphinxdoc and readthedocs, it would be practical to
+integrate these ADRs as part of our current documentation infrastructure.
+
+## Decision
+
+* We will use _Architecture Decision Records_ to document, propose and discuss
+any important or significant architectural and design decisions.
+
+* The ADR format will follow one of the two formats described in the sections below.
```

Comment:
It would be helpful to be slightly more specific in this reference to the sections below so the reader doesn't feel like they have to search for what this is referring to here.